smartsim/exp/ray/raycluster.py

```diff
@@ -0,0 +1,493 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import re
+import time as _time
+import uuid
+
+from ...entity import EntityList, SmartSimEntity
+from ...error import SSConfigError, SSUnsupportedError
+from ...settings import AprunSettings, QsubBatchSettings, SbatchSettings, SrunSettings
+from ...utils import get_logger
+from ...utils.helpers import expand_exe_path, init_default
+
+logger = get_logger(__name__)
+
+
+def delete_elements(dictionary, key_list):
+    """Delete elements from a dictionary.
+    :param dictionary: the dictionary from which the elements must be deleted.
+    :type dictionary: dict
+    :param key_list: the list of keys to delete from the dictionary.
+    :type key: any
+    """
+    for key in key_list:
+        if key in dictionary:
+            del dictionary[key]
+
+
+### TODO
```

Comment:
This can be taken out?

Reply:
Yeah, I left it till the last moment to make sure we did not forget any suggestion.